src/test/java/io/jenkins/plugins/analysis/core/graphs/HealthSeriesBuilderTest.java

```diff
@@ -0,0 +1,342 @@
+        when(STATIC_ANALYSIS_RUN_2.getTotalSize()).thenReturn(1);
+        when(STATIC_ANALYSIS_RUN_3.getTotalSize()).thenReturn(2);
+
+        SeriesBuilder builder = new HealthSeriesBuilder(HEALTH_DESCRIPTOR, resultTime);
+        CategoryDataset dataSet = builder.createDataSet(CONFIGURATION, RESULTS);
+
+        List<?> rowKeys = dataSet.getRowKeys();
+        List<?> colKeys = dataSet.getColumnKeys();
+        int rc = dataSet.getRowCount();
+        int cc = dataSet.getColumnCount();
+
+        assertThat(rc).isEqualTo(0);
+        assertThat(cc).isEqualTo(0);
+
+        assertThat(rowKeys.size()).isEqualTo(0);
+        assertThat(colKeys.size()).isEqualTo(0);
+    }
+
+    @Test
+    void createDataSetWithSampleResultsWhereConfigurationBuildCountDefinedIsTrueAndBuildCountIsTwo() {
+        ResultTime resultTime = mock(ResultTime.class);
+        when(resultTime.areResultsTooOld(CONFIGURATION, STATIC_ANALYSIS_RUN)).thenReturn(false);
+
+        when(CONFIGURATION.useBuildDateAsDomain()).thenReturn(false);
+        when(CONFIGURATION.isBuildCountDefined()).thenReturn(true);
+        when(CONFIGURATION.getBuildCount()).thenReturn(2);
+
+        when(STATIC_ANALYSIS_RUN.getBuild()).thenReturn(ANALYSIS_BUILD);
+        when(STATIC_ANALYSIS_RUN_2.getBuild()).thenReturn(ANALYSIS_BUILD_2);
+        when(STATIC_ANALYSIS_RUN_3.getBuild()).thenReturn(ANALYSIS_BUILD_3);
+
+        when(STATIC_ANALYSIS_RUN.getTotalSize()).thenReturn(0);
+        when(STATIC_ANALYSIS_RUN_2.getTotalSize()).thenReturn(1);
+        when(STATIC_ANALYSIS_RUN_3.getTotalSize()).thenReturn(2);
+
+        when(ANALYSIS_BUILD.compareTo(ANALYSIS_BUILD_2)).thenReturn(-1);
+        when(ANALYSIS_BUILD.compareTo(ANALYSIS_BUILD_3)).thenReturn(-1);
+
+        when(ANALYSIS_BUILD_2.compareTo(ANALYSIS_BUILD)).thenReturn(1);
+        when(ANALYSIS_BUILD_2.compareTo(ANALYSIS_BUILD_3)).thenReturn(-1);
+
+        when(ANALYSIS_BUILD_3.compareTo(ANALYSIS_BUILD)).thenReturn(1);
+        when(ANALYSIS_BUILD_3.compareTo(ANALYSIS_BUILD_2)).thenReturn(1);
+
+        when(HEALTH_DESCRIPTOR.isEnabled()).thenReturn(false);
+
+        SeriesBuilder builder = new HealthSeriesBuilder(HEALTH_DESCRIPTOR, resultTime);
+        CategoryDataset dataSet = builder.createDataSet(CONFIGURATION, RESULTS);
+
+        List<?> rowKeys = dataSet.getRowKeys();
+        List<?> colKeys = dataSet.getColumnKeys();
+
+        assertThat(rowKeys.size()).isEqualTo(1);
+        assertThat(colKeys.size()).isEqualTo(2);
+
+        int zero = dataSet.getValue(0, 0).intValue();
+        int one = dataSet.getValue(0, 1).intValue();
+
+        assertThat(zero).isEqualTo(0);
+        assertThat(one).isEqualTo(1);
+        assertThatThrownBy(() -> dataSet.getValue(0, 2)).isInstanceOf(IndexOutOfBoundsException.class);
+    }
+
+    @Test
+    void createDataSetWithSampleResultsWhereConfigurationUseBuildDateAsDomainIsTrue() {
+        SeriesBuilder builder = new HealthSeriesBuilder(HEALTH_DESCRIPTOR);
```

Comment:
Bei diesem Testfall fällt auf, dass recht viel Code dupliziert ist mit vorhergehenden und nachfolgenden Tests: Sinnvoller wäre das auslagern in Methoden.